sdk/go/dagger/client.go

```diff
@@ -62,7 +62,13 @@ func WithNoExtensions() ClientOpt {
 }
 
 // Connect to a Dagger Engine
-func Connect(ctx context.Context, opts ...ClientOpt) (*Client, error) {
+func Connect(ctx context.Context, opts ...ClientOpt) (_ *Client, rerr error) {
+	defer func() {
+		if rerr != nil {
```

Comment:
I fail to see where this `rerr` is set.
I will checkout the code and check my assumption and come back to this.

Reply:
I confirm that `rerr` is not set here.
Should we intercept any error in Connect and set `rerr` instead? Or just the `c.conn.Connect(ctx, cfg)`?

Reply:
I might be misunderstanding but `rerr` should be set whenever the second return val in non-nil anywhere in the func. E.g. https://go.dev/play/p/6vwRL72pmoV